src/algorithms/reco/JetReconstruction.cc

```diff
@@ -79,11 +80,13 @@ namespace eicrecon {
   }  // end 'init(std::shared_ptr<spdlog::logger>)'
 
 
-
-  template <typename T> std::unique_ptr<edm4eic::ReconstructedParticleCollection> JetReconstruction::process(const T* input_collection) {
-
-    // Store the jets
-    std::unique_ptr<edm4eic::ReconstructedParticleCollection> jet_collection { std::make_unique<edm4eic::ReconstructedParticleCollection>() };
+  void JetReconstruction::process(
+                                  const JetReconstruction::Input& input,
+                                  const JetReconstruction::Output& output
+                                 ) const {
```

Comment:
I touched on this offline, @johnlajoie , but one thing to consider is how to handle the templating here. I suppose a question for @veprbl , @wdconinc , and @nathanwbrei: can the algorithms interface handle templated Inputs? e.g.
```
template <typename T>
using JetReconstructionAlgorithm = algorithms::Algorithm<
      algorithms::Input<T>,
      algorithms::Output<edm4eic::ReconstructedParticle>
 >;
```

Because if possible, I'd prefer to keep the templating in place, especially with formal tracks coming in the near future...

Reply:
Alias templates are a thing https://en.cppreference.com/w/cpp/language/type_alias apparently

Reply:
I don't see why this shouldn't work. It's exactly the same as passing a `std::vector<T>` in the variadic input factories, https://github.com/eic/EICrecon/blob/main/src/algorithms/meta/CollectionCollector.h#L14.

Reply:
Implemented